File: packages/typescript-sdk/src/cache/noop.ts
/**
 * No-op cache provider that disables caching.
 *
 * Use this when you want to disable caching entirely.
 */

import type { CacheProvider } from "./interface.js";

/**
 * No-op cache provider.
 *
 * All get() calls return undefined (cache miss).
 * All set() calls do nothing.
 *
 * Use when you want to disable caching for testing,
 * debugging, or when caching isn't beneficial for your use case.
 *
 * @example
 * ```ts
 * import { LSBibleClient, NoopCacheProvider } from 'lsbible';
 *
 * const client = new LSBibleClient({
 *   cache: {
 *     provider: new NoopCacheProvider()
 *   }
 * });
 * ```
 */
export class NoopCacheProvider implements CacheProvider {
  async get<T>(_key: string): Promise<T | undefined> {
    return undefined;
  }

  async set<T>(_key: string, _value: T, _ttl: number): Promise<void> {
    // No-op
  }
}

File: packages/python-sdk/lsbible/cache/noop.py
"""No-op cache provider that disables caching."""

from typing import Any


class NoopCacheProvider:
    """No-op cache provider that disables caching entirely.

    All get() calls return None (cache miss), and all set() calls do nothing.
    Useful for debugging or situations where caching isn't beneficial.

    Example:
        ```python
        from lsbible import LSBibleClient, NoopCacheProvider

        # Disable caching for testing
        client = LSBibleClient(
            cache={"provider": NoopCacheProvider()}
        )
        ```
    """

    async def get(self, key: str) -> Any | None:
        """Get a value from the cache.

        Always returns None (cache miss).

        Args:
            key: Cache key (ignored)

        Returns:
            Always None
        """
        return None

    async def set(self, key: str, value: Any, ttl: int) -> None:
        """Set a value in the cache.

        Does nothing (no-op).

        Args:
            key: Cache key (ignored)
            value: Value to cache (ignored)
            ttl: Time to live in seconds (ignored)
        """
        pass
